apps/src/javalab/JavalabEditor.jsx

```diff
@@ -69,37 +81,47 @@ class JavalabEditor extends React.Component {
     };
   };
 
+  renameFileComplete = e => {
+    e.preventDefault();
+    this.props.renameFile(this.state.oldFilename, this.props.filename);
+    this.setState({renameFileActive: false});
+  };
+
+  enableRenameFile() {
+    this.setState({oldFilename: this.props.filename, renameFileActive: true});
+  }
+
   displayFileRename() {
     return (
       <div style={style.tabs}>
-        <div style={style.tab}>
+        <form style={style.renameForm} onSubmit={this.renameFileComplete}>
+          <div style={style.tab}>
+            <input
+              type="text"
+              value={this.props.filename}
+              onChange={e => {
+                this.props.setFilename(e.target.value);
+              }}
+            />
+          </div>
           <input
-            type="text"
-            ariaLabel="file name"
-            value={this.props.fileName}
-            onChange={e => this.props.setFileName(e.target.value)}
-            onBlur={() => this.setState({renameFileActive: false})}
+            className="btn btn-default btn-sm"
+            style={style.button}
+            type="submit"
+            value="Save"
```

Comment:
possibly future work, but user-facing strings should live in an i18n file

Reply:
👍 right now  the strings aren't final so I'm not putting them in i18n, but will add them when we finalize them.